### DISCUSSION TITLE
Add descriptions to secrets

### DISCUSSION THREAD

**Discussion by jamiebuilds** [community/community#15018] [Codespaces]
It would be really nice if secrets could have additional context added to them like a comment.

- "This secret is used for..."
- "Update this secret by..."
- "This secret is only to be used for..."
- etc

Yes its a good practice to name your secrets something self-descriptive, but sometimes additional context can be useful.

## Example

<img width="813" alt="Screen Shot 2022-04-15 at 1 13 16 PM" src="https://user-images.githubusercontent.com/952783/163627598-7af5882b-4c0c-4569-b9c9-de4eee21a403.png">

**Comment by nedbat** [comment]
Absolutely more context would be useful.  GitHub wants me to only use expiring tokens.  It would be great to put the URL to the token as part of the description of the secret, so that it's easy to regenerate them as needed.

**Comment by scott-doyland-burrows** [comment]
Descriptions for secrets is needed.

Also, it could be taken one step further, there could be a selector to chose common secrets, so I could choose a secret type of "AWS" and it would have a few boxes to enter AWS account name, AWS user name (not key), and a generic box for other info, and this would all be shown in clear text.

**Comment by mkindahl** [comment]
I agree with this, some additional information is needed in association with the secret. This is especially difficult when using company-wide secrets and at least these fields would be very useful to work in a company with multiple teams:

- **Who created the secret and when.** This is useful to check who to ask about the secret and what it is used for. Many times, secrets are created for temporary purposes but are useful in a wider context. Being able to reach the original person and figure out if it has wider utility is therefore useful. Also seeing if the secret is old or new makes it easier to decide if it should be removed or if it was replaced with a different secret that was created later.
- **The purpose of the secret.** Being able to tell what the purpose of the secret is make is easy to decide if it is appropriate for usage in a particular GitHub action. Having a generic description here would of course work for this.
- **Team that owns a secret.** Sometimes the owner is not the same as the creator and being able to tell what team or individual owns the secret is very useful in a larger organization.
- **When the secret expires.** Secrets almost always have an expiry date when they are no longer valid, interesting, or useful. Having this as a single field would also help searching for expired secrets and remove them.

**Comment by maxpatiiuk** [comment]
For anyone in the future, our workaround was to define a secret called `BEFORE_EDITING_ANY_SECRET_SEE_GITHUB_WORKFLOWS_README_MD_ON_PRODUCTION` (the name starts with `B` thus it's always near the top).

Note, all secrets must have a value. Otherwise, GitHub won't let you create them. Put anything as a value for this secret.

Then, create a `.github/workflows/README.md` on the production branch:

```html
# GitHub Secrets

Documentation for used GitHub secrets.

**On any changes to used secrets/tokens, update this file**

<table>
    <thead>
        <tr>
            <th>Name</th>
            <th>Active</th>
            <th>Description</th>
            <th>Used by</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>QODANA_TOKEN</td>
            <td>No</td>
            <td>
                Used by JetBrain's Qodana service
            </td>
            <td>
                <a href="https://github.com/specify/specify7/pull/2710">
                    #2710
                </a>
            </td>
        </tr>
        <tr>
            <td>WEBLATE_API_TOKEN</td>
            <td>Yes</td>
            <td>
                Regular weblate organization token. Used for API calls
            </td>
            <td>
                <a href="https://github.com/specify/specify7/blob/production/.github/workflows/test.yml">
                    test.yml
                </a>,
                <a href="https://github.com/specify/specify7/blob/weblate-localization/.github/workflows/push.yml">
                    push.yml
                </a>
            </td>
        </tr>
        <tr>
            <td>WEBLATE_PUSH_TO_GITHUB</td>
            <td>Yes</td>
            <td>
                Personal GitHub token (from @maxpatiiuk account). Personal token
                is used to bypass branch protection rules (to allow Weblate to
                push directly to production branch)
            </td>
            <td>
                <a href="https://github.com/specify/specify7/blob/weblate-localization/.github/workflows/push.yml">
                    push.yml
                </a>
            </td>
        </tr>
    </tbody>
</table>
```

Keep in mind though, that this file would be public, so don't put any sensitive information in there. On a positive side, people cloning your repository would have an idea of what secrets they need to add to get everything working
